src/Approximations/overapproximate.jl

```diff
@@ -1316,3 +1316,55 @@ function overapproximate(cap::Intersection{N,
                             ::Type{CartesianProductArray}, oa) where {N}
     overapproximate(Intersection(cap.Y, cap.X), oa)
 end
+
+"""
+    overapproximate(Z::Zonotope{N}, ::Type{<:Zonotope}, r) where {N<:Real}
+
+Reduce the order of a zonotope by overapproximating with a zonotope with less
+generators.
+
+### Input
+
+- `Z` -- zonotope
+- `Zonotope` -- desired type for dispatch
+- `r` -- desired order
+
+### Output
+
+A new zonotope with less generators, if possible.
+
+### Algorithm
+
+This function implements the algorithm described in A. Girard's
+*Reachability of Uncertain Linear Systems Using Zonotopes*, HSCC. Vol. 5. 2005.
+
+If the desired order is smaller than one, the zonotope is *not* reduced.
+"""
+function overapproximate(Z::Zonotope{N}, ::Type{<:Zonotope}, r) where {N<:Real}
```

Comment:
I wonder whether we should restrict `r` to type `Number` and/or add another alias method
`overapproximate(Z::Zonotope{N}, r::Number) where {N<:Real}`.

Reply:
> I wonder whether we should restrict r to type Number 

👍  here it makes sense for `r` to be either an integer or a rational, if we are going to add the type we can use a small union?

> add another alias method overapproximate(Z::Zonotope{N}, r::Number) where {N<:Real}

👎  the API of `overapproximate` requires the type of the expected set, we export `reduce_order` as a convenience function

Reply:
Okay for `r::Union{Integer, Rational}`.